docs/enhancements/sep-tarball-client.md

```diff
@@ -0,0 +1,170 @@
+consumer to implement. The structure of the tarball can change between
+versions and consumer implementations become fragile.
+
+This proposal aims to describe an API for a client that can consumers can use
+to extract data from a Sonobuoy tarball.
+
+## Objectives
+
+Provide a friendly API for easy access to data found in Sonobuoy Tarballs
+across versions. Users should already be fairly familiar with the API below
+as it might look a lot like client-go. Instead of querying an API server we
+are querying a Sonobuoy tarball.
+
+### Goals
+
+ - Define and implement an API to access files in a Sonobuoy tarball.
+ - Support all tarball versions starting at v0.9.0.
+ - Have a familiar API.
+
+### Non-Goals
+
+- We will be unable to customize the tarball for every plugin as people can
+write custom plugins. However, there should be an easy way for folks to get
+their results using this client. The main feature we are supporting is access
+to the core data generated by the [Sonobuoy queries][queries].
+- There will be no support for tarballs before Sonobuoy version 0.9.0.
+
+## Proposal
+
+```go
+type Tarball interface {
+    // Sonobuoy dumps stuff out that can be queried throught the k8s interface.
+    // The source will be a tarball instead of a k8s server.
+    Resources() kubernetes.Interface
+
+    // These are all the extra things that Sonobuoy picks up.
+    Hosts() []*Host
+    Metadata() *Metadata
+    Plugins() []*Plugin
+    PodLogs() []*PodLog
+    ServerVersion() *version.Info
+}
+
+// Host contains the metadata for a given host in a Kubernetes cluster.
+type Host struct {
+    // Name is the name of the host.
+    Name string
+    // Configz is the result of the configz endpoint.
+    KubeletConfig map[string]interface{}
+    // StatusCode is the result of the healthz endpoint for the node.
+    StatusCode int
+}
+
+// Metadata is all the stuff about the Sonobuoy run and how long it took to query the system.
+type Metadata struct {
+    // Config is the config used during this Sonobuoy run.
+    Config sonobuoy.Config
+    // QueryInfos shows how long each object took to query.
+    QueryInfos []*sonobuoy.QueryInfo
+}
+
+// Plugin is a single plugin that executed during a Sonobuoy run.
+type Plugin struct {
+    // Name is the name of the plugin.
+    Name string
+    // Results is a map of plugin result filenames to the file contents.
+    Results map[string][]byte
+}
+
+// PodLog are all the logs gathered from an individual pod.
+type PodLog struct {
+    // Name is the name of the pod.
+    Name string
+    // Logs are the logs in io.Reader form because they can be quite large.
+    Logs io.Reader
+}
+```
+
+Some sample code using this interface:
+
+```go
+data, _ := ioutil.ReadFile("sonobuoy.tar.gz")
+tarballClient := sonobuoy.NewTarball(data)
+
+// Getting resources (namespaced or not) should be really familiar.
+_ := tarballClient.Resources().Core().Nodes()
```

Comment:
I don't think we need to follow api conventions here.  There are cluster scoped and NS-scoped resources, it's much more akin to how RBAC's interface.